build.assets/tooling/cmd/notarize-apple-binaries/gonWrapper.go

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"os"
+	"path"
+
+	"github.com/gravitational/trace"
+	"github.com/hashicorp/go-hclog"
+	"github.com/mitchellh/gon/notarize"
+	"github.com/mitchellh/gon/package/zip"
+	"github.com/mitchellh/gon/sign"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	DevelopereIdentity string = "0FFD3E3413AB4C599C53FBB1D8CA690915E33D83"
+	BundleID           string = "com.gravitational.teleport"
+)
+
+type GonWrapper struct {
+	ctx    context.Context
+	logger hclog.Logger
+	config *GonConfig
+}
+
+func NewGonWrapper(config *GonConfig) *GonWrapper {
+	return &GonWrapper{
+		ctx:    context.Background(),
+		logger: NewHCLogLogrusAdapter(),
+		config: config,
+	}
+}
+
+func (gw *GonWrapper) SignAndNotarizeBinaries() error {
+	err := gw.SignBinaries()
+	if err != nil {
+		return trace.Wrap(err, "failed to sign binaries")
+	}
+
+	zipPath, err := gw.ZipBinaries()
+	if err != nil {
+		return trace.Wrap(err, "failed to zip binaries for notarization")
+	}
+	defer os.RemoveAll(path.Dir(zipPath))
+
+	logrus.Info("Signing and notarization complete!")
```

Comment:
Should you be using `gw.logger` here to log things?

Reply:
Good catch - fixed